src/socket/udp.rs

```diff
@@ -1,76 +1,68 @@
 use core::cmp::min;
-use managed::Managed;
+use managed::ManagedSlice;
 
 use {Error, Result};
 use wire::{IpProtocol, IpRepr, IpEndpoint, UdpRepr};
 use socket::{Socket, SocketMeta, SocketHandle};
-use storage::{Resettable, RingBuffer};
+use storage::RingBuffer;
 
-/// A buffered UDP packet.
-#[derive(Debug)]
-pub struct PacketBuffer<'a> {
+// Endpoint and size of an UDP packet.
+#[derive(Debug, Clone, Copy, Default)]
+pub struct PacketMetadata {
     endpoint: IpEndpoint,
-    size:     usize,
-    payload:  Managed<'a, [u8]>
+    payload_size: usize,
+    /// Dummy packets can be used to avoid wrap-arounds of packets in the payload buffer
+    dummy: bool,
 }
 
-impl<'a> PacketBuffer<'a> {
-    /// Create a buffered packet.
-    pub fn new<T>(payload: T) -> PacketBuffer<'a>
-            where T: Into<Managed<'a, [u8]>> {
-        PacketBuffer {
-            endpoint: IpEndpoint::default(),
-            size:     0,
-            payload:  payload.into()
-        }
-    }
+/// An UDP packet ring buffer.
+#[derive(Debug)]
+pub struct SocketBuffer<'a> {
```

Comment:
This has to have two lifetimes. If you think it can be one, try to construct multiple socket buffers without using `Vec`.